ToDoApp/ToDoApp/Respository/Respository.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Respository.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-22.
+//
+
+import Foundation
+import FirebaseDatabase
+import FirebaseDatabaseSwift
+protocol Repository {
+    func updateAndCreate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID
+    func read<T>() async -> Result<[T], Error> where T: Decodable
+}
+extension DatabaseReference: Repository {
+  // MARK: create a task
+    func updateAndCreate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID {
+        do {
+           try await self.child(value.id.uuidString).setValue(value.toDictionary)
+        } catch {
+            return TError.unsavedT
+        }
+        return nil
+    }
+    func read<T>() async -> Result<[T], Error> where T : Decodable {
+        var task: [T] = [T]()
+        self.observe(.value) { parentSnapshot in
+            guard let children = parentSnapshot.children.allObjects as? [DataSnapshot] else {
+                return
```

Comment:
all fallbacks should return errors

Reply:
I tried putting return error, but it didn't allow me to do so, it said "Unexpected non-void return value in void function"

Reply:
That's because you tried it not in the method body, but in the closure body which signature doesn't return result